src/osdc/Objecter.cc

```diff
@@ -3435,6 +3435,7 @@ void Objecter::handle_osd_op_reply(MOSDOpReply *m)
     // new tid
     s->ops.erase(op->tid);
     op->tid = ++last_tid;
+    s->ops[op->tid] = op; // add op to ops again
```

Comment:
yes, I surprised about this.... because we never meet this case before.

I searched the EAGAIN reply code in src/osd, and found the only source is from https://github.com/ceph/ceph/pull/15489/files#diff-fb41013d27e932534adb50eb3de2aaa5R1993